### ISSUE TITLE
Bounding box circles all the instances inside

### ISSUE THREAD

**Issue by remindchobits** [matterport/Mask_RCNN#1487]
Hi all,
I trained the model with own dataset and there is a problem while detecting. No matter how many instances in the test images, there is only one instance being detected, just like this

![result](https://user-images.githubusercontent.com/18598039/57187429-3a0c4f80-6f21-11e9-8a51-8201edc0742c.jpg)

The mask is relatively accurate, however the count of bbox is apparently wrong, it includes all masks in the image.
Has any one ever encountered this problem?
THX!

Here are my configurations:
Configurations:
BACKBONE                       resnet101
BACKBONE_STRIDES               [4, 8, 16, 32, 64]
BATCH_SIZE                     1
BBOX_STD_DEV                   [0.1 0.1 0.2 0.2]
COMPUTE_BACKBONE_SHAPE         None
DETECTION_MAX_INSTANCES        100
DETECTION_MIN_CONFIDENCE       0.7
DETECTION_NMS_THRESHOLD        0.3
FPN_CLASSIF_FC_LAYERS_SIZE     1024
GPU_COUNT                      1
GRADIENT_CLIP_NORM             5.0
IMAGES_PER_GPU                 1
IMAGE_CHANNEL_COUNT            3
IMAGE_MAX_DIM                  1024
IMAGE_META_SIZE                14
IMAGE_MIN_DIM                  800
IMAGE_MIN_SCALE                0
IMAGE_RESIZE_MODE              square
IMAGE_SHAPE                    [1024 1024    3]
LEARNING_MOMENTUM              0.9
LEARNING_RATE                  0.001
LOSS_WEIGHTS                   {'rpn_class_loss': 1.0, 'rpn_bbox_loss': 1.0, 'mrcnn_class_loss': 1.0, 'mrcnn_bbox_loss': 1.0, 'mrcnn_mask_loss': 1.0}
MASK_POOL_SIZE                 14
MASK_SHAPE                     [28, 28]
MAX_GT_INSTANCES               100
MEAN_PIXEL                     [123.7 116.8 103.9]
MINI_MASK_SHAPE                (56, 56)
NAME                           shapes
NUM_CLASSES                    2
POOL_SIZE                      7
POST_NMS_ROIS_INFERENCE        1000
POST_NMS_ROIS_TRAINING         2000
PRE_NMS_LIMIT                  6000
ROI_POSITIVE_RATIO             0.33
RPN_ANCHOR_RATIOS              [0.5, 1, 2]
RPN_ANCHOR_SCALES              (32, 64, 128, 256, 512)
RPN_ANCHOR_STRIDE              1
RPN_BBOX_STD_DEV               [0.1 0.1 0.2 0.2]
RPN_NMS_THRESHOLD              0.7
RPN_TRAIN_ANCHORS_PER_IMAGE    256
STEPS_PER_EPOCH                100
TOP_DOWN_PYRAMID_SIZE          256
TRAIN_BN                       False
TRAIN_ROIS_PER_IMAGE           100
USE_MINI_MASK                  True
USE_RPN_ROIS                   True
VALIDATION_STEPS               50
WEIGHT_DECAY                   0.0001

**Comment by buaacarzp** [comment]
hello,its my honer to meet the same question with you , my two instances what are same that was inside only one Bounging box circle, it let me fell so unhappy and agonizing. please help me or let us discuss this problem together,my qq number is 510695983

**Comment by remindchobits** [comment]
> hello,its my honer to meet the same question with you , my two instances what are same that was inside only one Bounging box circle, it let me fell so unhappy and agonizing. please help me or let us discuss this problem together,my qq number is 510695983

Hello!
Sorry for that I don't use qq, we can discuss it here and let more guys who meet the same problem join us.
Have you found any solution yet?

**Comment by buaacarzp** [comment]
Hey, I feel so happy that I receive your email .
Now, I find  the result of this phenomenon  my be the anchor box,  my picture's size  are 2048*2448, but the train data's size are 320*448, and the anchor box size are (8*6,16*6,32*6,64*6,128*6),  but  I can't consider it  more and more ,  I think  the anchor box's size is too big, so the anchor box cover my two data which are same, and give one classify and one P value.
whats your opinion on this phenomenon?
I am hoping your replay,thanks.


18811746117@163.com
 
From: remindchobits
Date: 2019-05-07 13:53
To: matterport/Mask_RCNN
CC: buaacarzp; Comment
Subject: Re: [matterport/Mask_RCNN] Bounding box circles all the instances inside (#1487)
hello,its my honer to meet the same question with you , my two instances what are same that was inside only one Bounging box circle, it let me fell so unhappy and agonizing. please help me or let us discuss this problem together,my qq number is 510695983
Hello!
Sorry for that I don't use qq, we can discuss it here and let more guys who meet the same problem join us.
Have you found any solution yet?

**Comment by remindchobits** [comment]
> Hey, I feel so happy that I receive your email . Now, I find the result of this phenomenon my be the anchor box, my picture's size are 2048*2448, but the train data's size are 320*448, and the anchor box size are (8*6,16*6,32*6,64*6,128*6), but I can't consider it more and more , I think the anchor box's size is too big, so the anchor box cover my two data which are same, and give one classify and one P value. whats your opinion on this phenomenon? I am hoping your replay,thanks. 18811746117@163.com From: remindchobits Date: 2019-05-07 13:53 To: matterport/Mask_RCNN CC: buaacarzp; Comment Subject: Re: [matterport/Mask_RCNN] Bounding box circles all the instances inside (#1487) hello,its my honer to meet the same question with you , my two instances what are same that was inside only one Bounging box circle, it let me fell so unhappy and agonizing. please help me or let us discuss this problem together,my qq number is 510695983 Hello! Sorry for that I don't use qq, we can discuss it here and let more guys who meet the same problem join us. Have you found any solution yet? — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or mute the thread.

Hi, thank you for the reply!
I've tried to turn the anchor box size to smaller numbers, my current RPN_ANCHOR_SCALES is (16, 32, 64, 128, 256), and I also turned the RPN_ANCHOR_RATIOS to (0.8,1,1.2) to generate square-like anchor boxes. However, this didn't work, the problem is still unsolved, and the detection accuracy is down
that really troubles me

**Comment by buaacarzp** [comment]
Hey,long time no see ,
I receive your email ,and you say that your anchor size is (16, 32, 64, 128, 256) , I think  the anchor box  is linked with the pic's depth,  for eg, if  there are two pictures ,and the size of them are 2448*2048， but the distance between the camera  and the thing which will be detection  are not same,  The distance between the object A and the camera  is 10cm , but the distance between the object B and the camera is 5cm, so the depth are not same , it will cause the anchor's size need to be changed.
So, now,  I guess  my object is small(because the distance is too far ), so I will turn down the value of the anchor box, about 2 hours, My model will be ready. 
If this measure will not be effective, I will try another way to solute it.


18811746117@163.com
 
From: remindchobits
Date: 2019-05-07 14:35
To: matterport/Mask_RCNN
CC: buaacarzp; Comment
Subject: Re: [matterport/Mask_RCNN] Bounding box circles all the instances inside (#1487)
Hey, I feel so happy that I receive your email . Now, I find the result of this phenomenon my be the anchor box, my picture's size are 20482448, but the train data's size are 320448, and the anchor box size are (86,166,326,646,128*6), but I can't consider it more and more , I think the anchor box's size is too big, so the anchor box cover my two data which are same, and give one classify and one P value. whats your opinion on this phenomenon? I am hoping your replay,thanks. 18811746117@163.com From: remindchobits Date: 2019-05-07 13:53 To: matterport/Mask_RCNN CC: buaacarzp; Comment Subject: Re: [matterport/Mask_RCNN] Bounding box circles all the instances inside (#1487) hello,its my honer to meet the same question with you , my two instances what are same that was inside only one Bounging box circle, it let me fell so unhappy and agonizing. please help me or let us discuss this problem together,my qq number is 510695983 Hello! Sorry for that I don't use qq, we can discuss it here and let more guys who meet the same problem join us. Have you found any solution yet? — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or mute the thread.
Hi, thank you for the reply!
I've tried to turn the anchor box size to smaller numbers, my current RPN_ANCHOR_SCALES is (16, 32, 64, 128, 256), and I also turned the RPN_ANCHOR_RATIOS to (0.8,1,1.2) to generate square-like anchor boxes. However, this didn't work, the problem is still unsolved, and the detection accuracy is down
that really troubles me

**Comment by buaacarzp** [comment]
I am so sorry, my experiment is so bad, it let me down, if you have another way to try ,please tell me ,thank you .

**Comment by remindchobits** [comment]
> I am so sorry, my experiment is so bad, it let me down, if you have another way to try ,please tell me ,thank you .

Hi,dude,
Now I have a conjecture about the problem, but I must verify something:
1. Are you using your own dataset which is labeled with labelme or other labeling tools ?
2. Is the problem only occurs in the bounding box instead of the masks ?

thanks!

**Comment by remindchobits** [comment]
> I am so sorry, my experiment is so bad, it let me down, if you have another way to try ,please tell me ,thank you .

Hi,
Problem solved.

**Comment by buaacarzp** [comment]
hey，are you ok now?  the matter had been solved? please tell me how to solve it. thank you.



18811746117@163.com
 
From: remindchobits
Date: 2019-05-09 09:29
To: matterport/Mask_RCNN
CC: buaacarzp; Comment
Subject: Re: [matterport/Mask_RCNN] Bounding box circles all the instances inside (#1487)
Closed #1487.

**Comment by remindchobits** [comment]
> hey，are you ok now? the matter had been solved? please tell me how to solve it. thank you. 18811746117@163.com From: remindchobits Date: 2019-05-09 09:29 To: matterport/Mask_RCNN CC: buaacarzp; Comment Subject: Re: [matterport/Mask_RCNN] Bounding box circles all the instances inside (#1487) Closed #1487. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or mute the thread.

Hi,

Yes, probably it's the problem of your label tool. You can view your json file and I guess you can't see any information about the bounding box, but while training, the code will automatically generate bounding box which circles all the labeled area inside no matter how many instances there are.

Currently for me there are 2 solutions

The first one is to remove all the training images which have more than one instance, which ensures that the automatically generated bounding boxes always circle the unique instance per image.

The second one is to rename your class with coco-like style, it's

Parent_child_num

or just transfer your dataset into coco format, and then try to train them with the coco.py instead of other scripts.

That's my solution, if anybody has better one pls share it to us xd

Thanks!

**Comment by Jer7** [comment]
> > hey，are you ok now? the matter had been solved? please tell me how to solve it. thank you. [18811746117@163.com](mailto:18811746117@163.com) From: remindchobits Date: 2019-05-09 09:29 To: matterport/Mask_RCNN CC: buaacarzp; Comment Subject: Re: [matterport/Mask_RCNN] Bounding box circles all the instances inside (#1487) Closed #1487. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or mute the thread.
> 
> Hi,
> 
> Yes, probably it's the problem of your label tool. You can view your json file and I guess you can't see any information about the bounding box, but while training, the code will automatically generate bounding box which circles all the labeled area inside no matter how many instances there are.
> 
> Currently for me there are 2 solutions
> 
> The first one is to remove all the training images which have more than one instance, which ensures that the automatically generated bounding boxes always circle the unique instance per image.
> 
> The second one is to rename your class with coco-like style, it's
> 
> Parent_child_num
> 
> or just transfer your dataset into coco format, and then try to train them with the coco.py instead of other scripts.
> 
> That's my solution, if anybody has better one pls share it to us xd
> 
> Thanks!

Hey!

If what you are saying is true then how do you explain that there is only one instance for each bbox in the balloon example ?

Thanks

**Comment by remindchobits** [comment]
> > > hey，are you ok now? the matter had been solved? please tell me how to solve it. thank you. [18811746117@163.com](mailto:18811746117@163.com) From: remindchobits Date: 2019-05-09 09:29 To: matterport/Mask_RCNN CC: buaacarzp; Comment Subject: Re: [matterport/Mask_RCNN] Bounding box circles all the instances inside (#1487) Closed #1487. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or mute the thread.
> > 
> > 
> > Hi,
> > Yes, probably it's the problem of your label tool. You can view your json file and I guess you can't see any information about the bounding box, but while training, the code will automatically generate bounding box which circles all the labeled area inside no matter how many instances there are.
> > Currently for me there are 2 solutions
> > The first one is to remove all the training images which have more than one instance, which ensures that the automatically generated bounding boxes always circle the unique instance per image.
> > The second one is to rename your class with coco-like style, it's
> > Parent_child_num
> > or just transfer your dataset into coco format, and then try to train them with the coco.py instead of other scripts.
> > That's my solution, if anybody has better one pls share it to us xd
> > Thanks!
> 
> Hey!
> 
> If what you are saying is true then how do you explain that there is only one instance for each bbox in the balloon example ?
> 
> Thanks

Since I didn't check the detail of the training data in balloon demo, I guess there is no problem with it, because after modifying the form of those data in my experiment the model can predict bounding box correctly. Just ensure each instance in the training data owns an unique bounding box
